lib/extensions/ar_region.rb

```diff
@@ -117,6 +117,16 @@ def partition_objs_by_remote_region(objs)
     def partition_ids_by_remote_region(ids)
       ids.partition { |id| self.id_in_current_region?(id) }
     end
+
+    def discover_my_region_number
+      region_id = connection.select_value("select min_value from miq_databases_id_seq") rescue nil
+      if region_id
+        id_to_region(region_id.to_i)
+      else
+        ENV.fetch("REGION", nil).try(:to_i) || File.read(File.join(Rails.root, "REGION")).to_i rescue 0
+      end
+    end
+    private :discover_my_region_number
```

Comment:
Prefer putting the private before the def like

```ruby
private def discover_my_region_number
```

or creating a private section

```ruby
private

def discover_my_region_number
```